src/main/java/umk/mat/pajda/ProjektZespolowy/configs/SecurityConfig.java

```diff
@@ -1,72 +1,83 @@
-import org.springframework.security.core.userdetails.User;
-import org.springframework.security.core.userdetails.UserDetails;
-import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.config.http.SessionCreationPolicy;
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.security.crypto.password.PasswordEncoder;
-import org.springframework.security.provisioning.InMemoryUserDetailsManager;
 import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+import umk.mat.pajda.ProjektZespolowy.services.AuthService;
 
 @Configuration
+@RequiredArgsConstructor
+@EnableWebSecurity
 public class SecurityConfig {
 
-  private String userPassword;
-
-  private String adminPassword;
+  private final JwtAuthenticationFilter jwtAuthenticationFilter;
+  private final AuthService authService;
 
   @Bean
-  @Profile("!tests")
-  public UserDetailsService userDetailsService() {
-    userPassword = System.getenv("USER_PASSWORD");
-    adminPassword = System.getenv("ADMIN_PASSWORD");
-    UserDetails user =
-        User.builder()
-            .username("user")
-            .password(passwordEncoder().encode(userPassword))
-            .roles("USER")
-            .build();
-
-    UserDetails admin =
-        User.builder()
-            .username("admin")
-            .password(passwordEncoder().encode(adminPassword))
-            .roles("ADMIN")
-            .build();
-
-    return new InMemoryUserDetailsManager(user, admin);
+  public AuthenticationProvider authenticationProvider() {
+    DaoAuthenticationProvider authenticationProvider = new DaoAuthenticationProvider();
+    authenticationProvider.setUserDetailsService(authService.userDetailsService());
+    authenticationProvider.setPasswordEncoder(passwordEncoder());
+    return authenticationProvider;
   }
 
   @Bean
   public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
     http.authorizeRequests(
             authorize ->
                 authorize
-                    .requestMatchers("/hello")
-                    .permitAll()
-                    .requestMatchers("/admin")
+                    .requestMatchers(
+                        HttpMethod.PATCH,
+                        "/user/editInformations",
+                        "/user/editPassword",
+                        "/user/editEmail")
+                    .authenticated()
+                    .requestMatchers(HttpMethod.DELETE, "/user")
+                    .authenticated()
+                    .requestMatchers(
+                        HttpMethod.GET,
```

Comment:
there is no requestMatchers for  HttpMethod.POST .    Add  requestMatchers  with  HttpMethod.POST for /review endpoint.